cloudencrypt/encode.go

```diff
@@ -0,0 +1,60 @@
+package cloudencrypt
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/gob"
+
+	"github.com/pkg/errors"
+)
+
+func encode(data map[string]string) ([]byte, error) {
+	var buffer bytes.Buffer
+
+	// Base64 encode
+	encoder := base64.NewEncoder(base64.StdEncoding, &buffer)
+
+	// Gzip compress
+	writer := gzip.NewWriter(encoder)
+
+	// gob encode
+	err := gob.NewEncoder(writer).Encode(data)
+	if err != nil {
+		return nil, errors.Wrapf(err, "gob encoder failed: %s", err.Error())
+	}
+
+	err = writer.Close()
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't close gzip writer: %s", err.Error())
+	}
+
+	err = encoder.Close()
+	if err != nil {
+		return nil, errors.Wrapf(err, "base64 encoder failed: %s", err.Error())
+	}
+
+	return buffer.Bytes(), nil
+}
+
+func decode(data []byte) (map[string]string, error) {
```

Comment:
Consider creating duplicate function `decodeString` ?

Reply:
Why?

Reply:
Ill chceck the code again. It made sense when I reviewed it as there were functions that were String and it had to be casted into `[]byte`

Reply:
It's not needed anymore